.github/workflows/node-4+.yml

```diff
@@ -10,6 +10,19 @@ permissions:
   contents: read
 
 jobs:
+  types:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ljharb/actions/node/install@main
+        name: 'npm install'
+        with:
+          skip-ls-check: true
+      # for some reason we've got to force typescript to install here
+      # even though the npm script has `typescript@latest`
+      - run: npm i --force typescript@latest
+      - run: npm run test-types
```

Comment:
this is "fine", but i would assume it'd be easier to add a prelint or postlint script that runs tsc?

Reply:
I'm not sure what you mean by this?

Reply:
what i mean is, type checking is a form of linting, and there's no need to make any workflow changes if a `prelint` or `postlint` script is defined in package.json, since it'll just automatically run as part of the pretest workflow.

Reply:
Ok but there's not a `lint` script (ESLint is run in `posttest`).

I can add this to `posttest` if that's what you mean/want, otherwise could you elaborate further on what you'd like?

Reply:
oh lol, true, this package is weird. in that case let's just move these lines to node-pretest.yml instead of the matrix workflow.